src/attention/utils/helpers.ts

```diff
@@ -1,98 +1,231 @@
+  arrowEl.style.transform = `rotate(${arrowRotation}deg)`
 }
 
 export async function useRecompute(state: AttentionState) {
-  if (!state.isShowing) return; // we're not currently showing the element, no reason to recompute
-  await state?.waitForDOM?.(); // wait for DOM to settle before computing
-  if (state.isCallout) return computeCalloutArrow(state); // we don't move the callout box, only its arrow
-  const position = await computePosition(
-    state.targetEl as ReferenceElement,
-    state.attentionEl as HTMLElement,
-    {
-      placement: state.directionName,
-      middleware: [
-        // Should we make this configurable, but have these as sane defaults?
-        flip(),
-        offset(8),
-        shift({ padding: 16 }),
-        // @ts-ignore
-        arrow({ element: state.noArrow ? undefined : state.arrowEl }), // FIXME
-      ],
+  if (!state?.isShowing) return // we're not currently showing the element, no reason to recompute
+  if (state?.waitForDOM) {
+    await state?.waitForDOM() // wait for DOM to settle before computing
+  }
+  if (state?.isCallout) return computeCalloutArrow(state)
+  
+  if (!state?.targetEl || !state?.attentionEl) return
+  
+  const targetEl: ReferenceElement = state?.targetEl
+  const attentionEl: HTMLElement = state?.attentionEl
+  
+  computePosition(targetEl, attentionEl, {
+    placement: state?.directionName ?? 'bottom',
+    middleware: [
+      offset({ mainAxis: state?.distance ?? 8, crossAxis: state?.skidding ?? 0}), // offers flexibility over how to place the attentionEl towards its targetEl both on the x and y axis (horizontally and vertically).
+      state?.flip && flip({ //when flip is set to true it will move the attentionEl's placement to its opposite side or to the preferred placements if fallbackPlacements has a value
+        fallbackAxisSideDirection: 'start', // the preferred placement axis fit when flip is set to true and fallbackPlacements does not have a value. 'start' represents 'top' or 'left'.
+        fallbackPlacements: state?.fallbackPlacements
+      }),
+      shift({ padding: 16}),
+      !state?.noArrow && state?.arrowEl && arrow({ element: state?.arrowEl }),
+    ],
+  }).then(({ x, y, middlewareData, placement }) => {
+    state.actualDirection = placement
+    
+    Object.assign(attentionEl?.style, {
+      left: `${x}px`,
+      top: `${y}px`,
+    })
+    
+    const side: Directions = placement.split('-')[0] as Directions
+    const staticSide: Directions = opposites[side]
+    const isRtl = window.getComputedStyle(attentionEl).direction === 'rtl' //checks whether the text direction of the attentionEl is right-to-left. Helps to calculate the position of the arrowEl and ensure proper alignment 
+    const arrowDirection: Directions = opposites[placement]
+    const arrowPlacement: string = arrowDirection.split('-')[1]
+    const arrowRotation: number = rotation[arrowDirection]
+    
+    if (middlewareData.arrow && state?.arrowEl) {
```

Comment:
Maybe safeguard `middlewareData?.arrow`?

Reply:
Updated in this commit: [a7fe061](https://github.com/warp-ds/core/pull/15/commits/a7fe061a8bd466af2a2fa382079595ea988beb9d)